cmd/infra/aws/route53.go

```diff
@@ -0,0 +1,206 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/aws/aws-sdk-go/service/route53/route53iface"
+	"github.com/pkg/errors"
+)
+
+func (o *CreateInfraOptions) LookupPublicZone(client route53iface.Route53API) (string, error) {
+	name := o.BaseDomain
+	id, err := lookupZone(client, name, false)
+	if err != nil {
+		log.Error(err, "Public zone not found", "name", name)
+		return "", err
+	}
+	log.Info("Found existing public zone", "name", name, "id", id)
+	return id, nil
+}
+
+func lookupZone(client route53iface.Route53API, name string, isPrivateZone bool) (string, error) {
+	var res *route53.HostedZone
+	f := func(resp *route53.ListHostedZonesOutput, lastPage bool) (shouldContinue bool) {
+		for idx, zone := range resp.HostedZones {
+			if zone.Config != nil && isPrivateZone == aws.BoolValue(zone.Config.PrivateZone) && strings.TrimSuffix(aws.StringValue(zone.Name), ".") == strings.TrimSuffix(name, ".") {
+				res = resp.HostedZones[idx]
+				return false
+			}
+		}
+		return !lastPage
+	}
+	if err := client.ListHostedZonesPages(&route53.ListHostedZonesInput{}, f); err != nil {
+		return "", err
+	}
+	if res == nil {
+		return "", errors.Errorf("Hosted zone %s not found", name)
+	}
+	return cleanZoneID(*res.Id), nil
+}
+
+func (o *CreateInfraOptions) CreatePrivateZone(client route53iface.Route53API, vpcID string) (string, error) {
+	name := fmt.Sprintf("%s.%s", o.Name, o.BaseDomain)
+	id, err := lookupZone(client, name, true)
```

Comment:
nit: can inline the if sentence here since variables are only used within its context